lib/matobj.gi

```diff
@@ -538,26 +538,30 @@ InstallMethod( CopySubVector,
     CopySubVector(dst,dcols,src,scols);
 end );
 
-InstallMethod( Randomize,
-  "generic method for a vector",
-  [ IsVectorObj and IsMutable ],
-  function(vec)
+InstallMethodWithRandomSource( Randomize,
+  "for a random source and a vector object",
+  [ IsRandomSource, IsVectorObj and IsMutable ],
+  function( rs, vec )
     local basedomain, i;
     basedomain := BaseDomain( vec );
     for i in [ 1 .. Length( vec ) ] do
-        vec[ i ] := Random( basedomain );
+        vec[ i ] := Random( rs, basedomain );
     od;
+    return vec;
 end );
 
-InstallMethod( Randomize,
-  "generic method for a vector and a random source",
-  [ IsVectorObj and IsMutable, IsRandomSource ],
-  function(vec, rs)
-    local basedomain, i;
-    basedomain := BaseDomain( vec );
-    for i in [ 1 .. Length( vec ) ] do
-        vec[ i ] := Random( rs, basedomain );
+InstallMethodWithRandomSource( Randomize,
+  "for a random source and a matrix object",
+  [ IsRandomSource, IsMatrixObj and IsMutable ],
```

Comment:
The code this replaces was using `IsVectorObj`, and now here is `IsMatrixObj`? Am I missing something? Aha, perhaps this is simply a new function that was not present before, and the diff is therefore misleading? In that case, it would be nice if the commit message / PR description explicitly mentioned this new functionality (but it's not super important either, it's just a suggestion).